lib/sftp/shell.rb

```diff
@@ -2,7 +2,11 @@ module SFTP
   module Shell
     def self.run(array_of_commands)
       command_str = [array_of_commands].join(" ")
-      `bash  -c \"#{command_str}\"`
+      output = `bash  -c \"#{command_str}\"`
+      unless $CHILD_STATUS.success?
+        raise Error, "Error occurred during SFTP process: #{$CHILD_STATUS.exitstatus}"
+      end
+      output
```

Comment:
I changed this to return the standard out output of the command. That's important for the `ls` command.

Reply:
Ah, I missed that, thanks. Good you added the test to catch the regression.